enhancements/declarative-index-config.md

```diff
@@ -865,34 +865,22 @@ $ tree
 ├── etcd.json
 ├── amqstreams.json
 ```
-#### Editing a package inside an index
+#### Creating new indexes using the config files/Editing a package inside an existing index
 
-A new sub-command `update` will be introduced under `opm index`, which will take a folder containing json representations of packages as it's input along with an existing index image tag, and will replace the edited config files in the index container. 
+A new sub-command `build` will be introduced under `opm index`, which will take a folder containing json representations of packages as it's input, and will build a container image with the folder and the `opm` binary in it. 
 
 ```bash
 $ opm index inspect --image=quay.io/some-namespace/my-index 
 $ // edit a my-index/package-config.json 
-$ opm index update my-index --tag=docker.io/some-namespace/my-index:latest
-``` 
-
-> Note: For a seamless UX, the following things should be consider during implementation when the `opm index update` command is summoned: 
-> 1. Validate the files inside the folder given as input to the command to ensure they are valid json representation of packages (and not random files with random content, which could be potential security risks) 
-> 2. If the files are valid, take a git diff between the present content of the index (i.e the existing json representations) and the content of the folder given as input. Display the diff as an output of the command with the following message 
-> "The following changes will be updated in the index:"
-> If there is no diff between the content in the container image and the content of the local folder, throw an error with the message "No diff found between index and local content to update."    
-
-
-#### Creating new indexes using the config files from an existing index
-
-A new sub-command `create` will be introduced under `opm index`, that will take a folder containing json/yaml representation of packages, and a container image tag as input. The sub-command will create a new index container, walk the tree of files to find the json representations and store them in the container. 
-
-```bash
+$ opm index build --configs=my-index --tag=docker.io/some-namespace/my-index:latest
 $ tree community-operators 
 community-operators
 ├── amqstreams.json
 └── etcd.json
-$ opm index create --from=community-operators --tag=docker.io/some-namespace/community-operators
-```   
+$ opm index build --from=community-operators --tag=docker.io/some-namespace/community-operators
+```
+
+> Note: The `build` command should also validate the files inside the folder given as input to the command to ensure they are valid json representation of packages (and not random files with random content, which could be potential security risks) before building the index image.      
```

Comment:
> coupling between opm and the dependency resolver

I don't think so. Things should be able to consume index content for whatever purpose without expecting the definition of a valid index to change.

It seems reasonable to me to have validations associated with blobs per "schema" that do things like flag package entries with identical names or bundle entries referencing a package entry that is missing.

> thoughts

If one of the goals of this proposal is to allow index maintainers to operate on a declarative textual representation as their source of truth, then I don't understand the advantages of having validation operate on container images (besides consistency with the "index" subcommands) rather than on the declarative text file(s) directly. Verifying that registry server endpoints return appropriate things based on the content of a declarative index config seems to me like a responsibility of the registry server tests.

Reply:
>If one of the goals of this proposal is to allow index maintainers to operate on a declarative textual representation as their source of truth, then I don't understand the advantages of having validation operate on container images (besides consistency with the "index" subcommands) rather than on the declarative text file(s) directly.

@benluddy are you suggesting a `opm registry validate --configs` command *instead* of performing this validation here before pushing bundles to the config?